### ISSUE TITLE
Multiple ServiceEntries with the same host cause traffic to be blackholed

### ISSUE THREAD

**Issue by jessebye** [istio/istio#51466]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

If two `ServiceEntries` are created that share the same host but different port/resolution configuration, the traffic is blackholed. Deleting either of the `ServiceEntry` resources resolves the problem and allows traffic to egress as expected.

Entry 1:
```yaml
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: dep1
spec:
  exportTo:
  - .
  hosts:
  - api.twilio.com
  location: MESH_EXTERNAL
  ports:
  - name: https
    number: 443
    protocol: TLS
```

Entry 2:
```yaml
apiVersion: networking.istio.io/v1
kind: ServiceEntry
metadata:
  name: dep2
spec:
  exportTo:
  - .
  hosts:
  - api.twilio.com
  location: MESH_EXTERNAL
  ports:
  - name: https
    number: 443
    protocol: HTTPS
  resolution: DNS
```

It seems like the entries are merged but in an unexpected way, which results in a bad cluster configuration in Envoy which blocks egress.

### Version

```Text
❯ istioctl version
client version: 1.22.1
control plane version: 1.22.0
data plane version: 1.22.0 (873 proxies)
❯ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.1
Kustomize Version: v4.5.4
Server Version: v1.29.4-eks-036c24b
```

**Comment by howardjohn** [comment]
there was some recent fixes around this, might be worth trying 1.22.1. Not sure they were fixed in that version though (on my phone so hard to check)

**Comment by jessebye** [comment]
@howardjohn thanks for the suggestion, I'm going to test with 1.22.1 and will let you know if this is resolved 👍

**Comment by howardjohn** [comment]
Thanks for the update. #50711 was the fix I was thinknig of but its on 1.22.0 so no impact here, sorry for the red-herring.

**Comment by jessebye** [comment]
not stale
